client/ayon_houdini/api/hda_utils.py

```diff
@@ -506,13 +508,85 @@ def _resolve_entity_uri(entity_uri: str, resolve_roots: bool = False):
     return response.data[0]["entities"]
 
 
+def _resolve_one_entity_uri(entity_uri: str) -> dict:
+    """Resolve an entity URI to a single entity.
+
+    Args:
+        entity_uri (str): The entity URI to resolve.
+
+    Raises:
+        ValueError: If the entity could not be resolved with input values.
+
+    """
+    try:
+        entities = _resolve_entity_uri(entity_uri)
+    except RuntimeError as exc:
+        raise ValueError(exc)
+
+    if not entities:
+        raise ValueError(f"Unable to find a matching entity for {entity_uri}")
+
+    if len(entities) > 1:
+        raise ValueError(
+            f"Multiple entities found for {entity_uri}: {entities}"
+        )
+
+    return entities[0]
+
+
+def get_version_id(
+    project_name: str,
+    folder_path: str,
+    product_name: str,
+    version: str,
+) -> str:
+    """Get version id.
+    """
+    entity_uri: str = _construct_ayon_entity_uri_str(
+        project_name=project_name,
+        folder_path=folder_path,
+        product=product_name,
+        version=version,
+        representation_name="",
+    )
+    version_from_uri = _resolve_one_entity_uri(entity_uri)
+    return version_from_uri["versionId"]
+
+
+def get_version(
+    project_name: str,
+    folder_path: str,
+    product_name: str,
+    version: str,
+) -> dict:
+    """Get version entity.
+
+    Args:
+        project_name (str): Project name.
+        folder_path (str): Folder name.
+        product_name (str): Product name.
+        version (str): Version name as string.
+
+    Returns:
+        dict: Version entity.
+
+    """
+    version_id = get_version_id(
+        project_name,
+        folder_path,
+        product_name,
+        version,
+    )
+    return get_version_by_id(project_name, version_id)
```

Comment:
These are best reversed, or the latter should not call the first. Now you're querying the database twice.

But the other way you don't with:
```python
get_version_id(...):
   get_version(...)["versionId"]
```

Reply:
Ah nevermind - that's wrong because the resolve endpoint doesn't return the actual entity, just the relevant ids 🤔